stl/inc/ranges

```diff
@@ -6842,6 +6843,472 @@ namespace ranges {
+        noexcept((noexcept(static_cast<_ResultType>(
+            _STD declval<const _LHSTupleTypes&>() - _STD declval<const _RHSTupleTypes&>())) && ...)) {
+        // clang-format on
+        const auto _Get_smallest_distance_closure = [&_Lhs_tuple, &
+            _Rhs_tuple ]<std::size_t _FirstIdx /* = 0 */, std::size_t... _Idxs>(index_sequence<_FirstIdx, _Idxs...>) {
+            const _ResultType _First_size = static_cast<_ResultType>(_STD get<0>(_Lhs_tuple) - _STD get<0>(_Rhs_tuple));
+
+            if (_First_size == 0) {
+                return static_cast<_ResultType>(0);
+            }
+            const initializer_list<_ResultType> _Sizes = {
+                _First_size, (_STD get<_Idxs>(_Lhs_tuple) - _STD get<_Idxs>(_Rhs_tuple))...};
+            return _First_size < 0 ? static_cast<_ResultType>((_RANGES max)(_Sizes))
+                                   : static_cast<_ResultType>((_RANGES min)(_Sizes));
+        };
+
+        return _Get_smallest_distance_closure(index_sequence_for<_LHSTupleTypes...>{});
+    }
+
+    template <class... _LHSTupleTypes, class... _RHSTupleTypes>
+        requires(sizeof...(_LHSTupleTypes) == sizeof...(_RHSTupleTypes))
+    _NODISCARD constexpr bool _Zip_iterator_sentinel_equal(
+        const tuple<_LHSTupleTypes...>& _Lhs_tuple, const tuple<_RHSTupleTypes...>& _Rhs_tuple)
+        // clang-format off
+        noexcept((noexcept(_STD declval<const _LHSTupleTypes&>() == _STD declval<const _RHSTupleTypes&>()) && ...)) {
+        // clang-format on
+        const auto _Evaluate_equality_closure =
+            [&_Lhs_tuple, &_Rhs_tuple ]<size_t... _Indices>(index_sequence<_Indices...>) {
+            return ((_STD get<_Indices>(_Lhs_tuple) == _STD get<_Indices>(_Rhs_tuple)) || ...);
+        };
+
+        return _Evaluate_equality_closure(index_sequence_for<_LHSTupleTypes...>{});
+    }
+
+    // clang-format off
+    _EXPORT_STD template <input_range... _ViewTypes>
+        requires (view<_ViewTypes> && ...) && (sizeof...(_ViewTypes) > 0)
+    class zip_view : public view_interface<zip_view<_ViewTypes...>> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ tuple<_ViewTypes...> _Views;
+
+        template <bool _IsConst>
+        struct _Category_base {};
+
+        template <bool _IsConst>
+            requires _All_forward<_IsConst, _ViewTypes...>
+        struct _Category_base<_IsConst> {
+            using iterator_category = input_iterator_tag;
+        };
+
+        template <bool _IsConst>
+        class _Iterator : public _Category_base<_IsConst> {
+        private:
+            /* [[no_unique_address]] */ tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...> _Current;
+
+            friend zip_view;
+
+            constexpr explicit _Iterator(tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...> _Current_) noexcept(
+                is_nothrow_move_constructible_v<
+                    tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...>>) // strengthened

```

Comment:
Suggestion: `tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...>` occurs 3 times and is verbose enough to cause wrapping - extracting a `_Current_tuple` or `_My_tuple` typedef (or any other name) would help readability.

Reply:
Actually, I *did* have a private type alias for that at one point, but it was suggested in other code reviews that it be removed. I still do believe that using a type alias is the right choice, so if you agree with that, then I'll add it back. (Sorry, @strega-nil-ms!)

Reply:
Thanks (I hadn't read the entire history before reviewing). I tracked down the original comment to make sure we weren't missing anything - it was https://github.com/microsoft/STL/pull/3035#discussion_r955450548 where @strega-nil-ms said "This seems unnecessary." While an alias isn't absolutely necessary here, I believe that extracting it improves clarity - both by reducing visual clutter and avoiding the need to verify that the exact same type is being formed every time.